lib/page/settings_page/setting_confirmation.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_platform_widgets/flutter_platform_widgets.dart';
+
+/// Holds information for showing a confirmation dialog.
+class SettingConfirmation {
+  /// The confirmation title.
+  final String title;
+  /// The confirmation content.
+  final String content;
+  
+
+  SettingConfirmation({required this.title, required this.content});
+}
+
+Future<bool> showConfirmationDialog(BuildContext context, SettingConfirmation confirmation) {
+  return showPlatformDialog(
+    context: context,
+    builder: (context) => PlatformAlertDialog(
+      title: Text(confirmation.title),
+      content: Text(confirmation.content),
+      actions: [
+        PlatformDialogAction(
+          child: PlatformText("Cancel"),
+          onPressed: () {
+            Navigator.of(context).pop(false);
+          },
+        ),
+        PlatformDialogAction(
+          child: PlatformText("Ok"),
+          onPressed: () {
+            Navigator.of(context).pop(true);
+          },
+        ),
+      ],
+    ),
+  ) as Future<bool>;
```

Comment:
`showConfirmationDialog` should be async, and then `showPlatformDialog` should be awaited inside the function.

```dart
Future<bool> showConfirmationDialog(...) async {
  return await showPlatformDialog(
    ...
  );
}
```
This likely does not require casting as dart should infer the type parameter of `showPlatformDialog` based on the return type of `showConfirmationDialog`. In the case that it does require casting, you would first await `showPlatformDialog` to resolve the `Future<dynamic>` to just `dynamic`, then cast that to a `bool`.